File: exams/villagomez/u3/examExercise/model/BubbleSort.js
export class BubbleSort {
    sort(array) {
        let n = array.length;
        let swapped;
        for (let i = 0; i < n - 1; i++) {
            swapped = false;
            for (let j = 0; j < n - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    [array[j], array[j + 1]] = [array[j + 1], array[j]]; // Swap
                    swapped = true;
                }
            }
            if (!swapped) break;
        }
        return array;
    }
}

File: hw/travez/u3/HW30Strategy/hw30Strategy/src/main/java/espe/edu/ec/SortApp/model/BubbleSort.java
package espe.edu.ec.SortApp.model;

import espe.edu.ec.SortApp.controller.SortingStrategy;

/**
 *
 * @author Ismael Travez, TheJavaBandits, DCCO-ESPE
 */
public class BubbleSort implements SortingStrategy {

    @Override
    public int[] sort(int[] array) {
        int n = array.length;
        boolean swapped;
        for (int i = 0; i < n - 1; i++) {
            swapped = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    // Swap array[j] and array[j + 1]
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                    swapped = true;
                }
            }
            // If no elements were swapped, the array is already sorted
            if (!swapped) {
                break;
            }
        }
        return array;
    }
}
